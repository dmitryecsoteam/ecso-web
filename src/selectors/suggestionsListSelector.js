export default (cities, value) => {

    const inputValue = value.trim().toLowerCase();
    const inputLength = inputValue.length;

    return inputLength === 0 ? [] : cities.filter(city =>
        (
            city.name.some((name) => (
                name.toLowerCase().startsWith(inputValue)
            ))
        )
    ).sort(city => city.nameEn.toLowerCase().startsWith(inputValue) ? -1 : 1);
};