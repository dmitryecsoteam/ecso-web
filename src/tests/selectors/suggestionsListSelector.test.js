import suggestionsListSelector from '../../selectors/suggestionsListSelector';
import cities from '../fixtures/destinations';

test('should return empty array on empty input string', () => {
    const suggestions = suggestionsListSelector(cities, '');
    expect(suggestions).toEqual([]);
});

test('should return empty array when no match', () => {
    const suggestions = suggestionsListSelector(cities, 'ะช');
    expect(suggestions).toEqual([]);
});

test('should return list of suggestions', () => {
    const suggestions = suggestionsListSelector(cities, 'ok');
    expect(suggestions).toMatchSnapshot();
});