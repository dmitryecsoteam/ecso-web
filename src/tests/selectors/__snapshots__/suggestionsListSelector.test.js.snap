// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return list of suggestions 1`] = `
Array [
  Object {
    "_id": 6,
    "countryEn": "Japan",
    "name": Array [
      "Okayama",
      "Окаяма",
      "岡山市",
      "OKJ",
    ],
    "nameEn": "Okayama",
  },
  Object {
    "_id": 7,
    "countryEn": "Japan",
    "name": Array [
      "Ok",
    ],
    "nameEn": "Ok",
  },
  Object {
    "_id": 5,
    "countryEn": "Japan",
    "name": Array [
      "Naha",
      "Наха",
      "那覇市",
      "OKA",
    ],
    "nameEn": "Naha",
  },
  Object {
    "_id": 4,
    "countryEn": "Japan",
    "name": Array [
      "Sapporo",
      "Саппоро",
      "札幌市",
      "SPK",
      "CTS",
      "OKD",
    ],
    "nameEn": "Sapporo",
  },
]
`;
